pkg/oidc/token_test.go

```diff
@@ -0,0 +1,214 @@
+
+	assert.Equal(t, claims.Issuer, tokenClaimsData.GetIssuer())
+	assert.Equal(t, claims.Subject, tokenClaimsData.GetSubject())
+	assert.Equal(t, []string(claims.Audience), tokenClaimsData.GetAudience())
+	assert.Equal(t, claims.Expiration.AsTime(), tokenClaimsData.GetExpiration())
+	assert.Equal(t, claims.IssuedAt.AsTime(), tokenClaimsData.GetIssuedAt())
+	assert.Equal(t, claims.Nonce, tokenClaimsData.GetNonce())
+	assert.Equal(t, claims.AuthTime.AsTime(), tokenClaimsData.GetAuthTime())
+	assert.Equal(t, claims.AuthorizedParty, tokenClaimsData.GetAuthorizedParty())
+	assert.Equal(t, claims.SignatureAlg, tokenClaimsData.GetSignatureAlgorithm())
+	assert.Equal(t, claims.AuthenticationContextClassReference, tokenClaimsData.GetAuthenticationContextClassReference())
+
+	claims.SetSignatureAlgorithm(jose.ES384)
+	assert.Equal(t, jose.ES384, claims.SignatureAlg)
+}
+
+func TestNewAccessTokenClaims(t *testing.T) {
+	want := &AccessTokenClaims{
+		TokenClaims: TokenClaims{
+			Issuer:     "zitadel",
+			Subject:    "hello@me.com",
+			Audience:   Audience{"foo"},
+			Expiration: 12345,
+			JWTID:      "900",
+		},
+	}
+
+	got := NewAccessTokenClaims(
+		want.Issuer, want.Subject, nil,
+		want.Expiration.AsTime(), want.JWTID, "foo", time.Second,
+	)
+
+	// Make equal not fail on dynamic timestamp
+	got.IssuedAt = 0
+	got.NotBefore = 0
+
+	assert.Equal(t, want, got)
+}
+
+func TestIDTokenClaims_GetAccessTokenHash(t *testing.T) {
+	assert.Equal(t, idTokenData.AccessTokenHash, idTokenData.GetAccessTokenHash())
+}
+
+func TestIDTokenClaims_SetUserInfo(t *testing.T) {
+	want := IDTokenClaims{
+		TokenClaims: TokenClaims{
+			Subject: userInfoData.Subject,
+		},
+		UserInfoProfile: userInfoData.UserInfoProfile,
+		UserInfoEmail:   userInfoData.UserInfoEmail,
+		UserInfoPhone:   userInfoData.UserInfoPhone,
+		Address:         userInfoData.Address,
+	}
+
+	var got IDTokenClaims
+	got.SetUserInfo(userInfoData)
+
+	assert.Equal(t, want, got)
+}
+
+func TestNewIDTokenClaims(t *testing.T) {
+	want := &IDTokenClaims{
+		TokenClaims: TokenClaims{
+			Issuer:                              "zitadel",
+			Subject:                             "hello@me.com",
+			Audience:                            Audience{"foo", "just@me.com"},
+			Expiration:                          12345,
+			AuthTime:                            12000,
+			Nonce:                               "6969",
+			AuthenticationContextClassReference: "something",
+			AuthenticationMethodsReferences:     []string{"some", "methods"},
+			AuthorizedParty:                     "just@me.com",
+			ClientID:                            "just@me.com",
+		},
+	}
+
+	got := NewIDTokenClaims(
+		want.Issuer, want.Subject, want.Audience,
+		want.Expiration.AsTime(),
+		want.AuthTime.AsTime().Add(time.Second),
+		want.Nonce, want.AuthenticationContextClassReference,
+		want.AuthenticationMethodsReferences, want.AuthorizedParty,
+		time.Second,
+	)
+
+	// Make equal not fail on dynamic timestamp
+	got.IssuedAt = 0
```

Comment:
same as just above

Reply:
done, github doesn't mark as outdated because it was solved on the lines above.